oauth-common/src/main/java/io/strimzi/kafka/oauth/common/HttpUtil.java

```diff
@@ -0,0 +1,121 @@
+        return postOrGet(uri, socketFactory, verifier, authorization, contentType, body, responseType);
+    }
+
+    public static <T> T postOrGet(URI uri, SSLSocketFactory socketFactory, HostnameVerifier hostnameVerifier, String authorization, String contentType, String body, Class<T> responseType) throws IOException {
+        HttpURLConnection con;
+        try {
+            con = (HttpURLConnection) uri.toURL().openConnection();
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("Malformed token endpoint url: " + uri);
+        }
+
+        if (con instanceof HttpsURLConnection) {
+            HttpsURLConnection scon = (HttpsURLConnection) con;
+            if (socketFactory != null) {
+                scon.setSSLSocketFactory(socketFactory);
+            }
+            if (hostnameVerifier != null) {
+                scon.setHostnameVerifier(hostnameVerifier);
+            }
+        } else if (socketFactory != null) {
+            log.warn("SSL socket factory set but url scheme not https ({})", uri);
+        }
+
+        con.setUseCaches(false);
+        if (body != null) {
+            con.setDoOutput(true);
+        }
+
+        con.setRequestMethod(body != null ? "POST" : "GET");
+        if (authorization != null) {
+            con.setRequestProperty("Authorization", authorization);
+        }
+        con.setRequestProperty("Accept", "application/json");
+        if (body != null && body.length() > 0) {
+            if (contentType == null) {
+                throw new IllegalArgumentException("contentType must be set when body is not null");
+            }
+            con.setRequestProperty("Content-Type", contentType);
+        }
+
+        try {
+            con.connect();
+        } catch (ConnectException e) {
+            throw new IOException("Failed to connect to: " + uri, e);
+        }
+
+        if (body != null && body.length() > 0) {
+            OutputStream out = con.getOutputStream();
+            out.write(body.getBytes(StandardCharsets.UTF_8));
+            out.close();
+        }
+
+        int code = con.getResponseCode();
+        if (code != 200) {
+            InputStream err = con.getErrorStream();
+            if (err != null) {
+                ByteArrayOutputStream errbuf = new ByteArrayOutputStream(4096);
+                try {
+                    copy(err, errbuf);
+                } catch (Exception e) {
+                    log.warn("[IGNORED] Failed to read response body", e);
+                }
+
+                throw new RuntimeException("Request failed with status " + code + ": " + errbuf.toString(StandardCharsets.UTF_8.name()));
+            } else {
+                throw new RuntimeException("Request failed with status " + code + " " + con.getResponseMessage());
+            }
+        }
+        return JSONUtil.readJSON(con.getInputStream(), responseType);
```

Comment:
This one here is also relying on other people's code being properly implemented. Will protect it ...